### ISSUE TITLE
Change how the names of timestamp files are generated.

### ISSUE THREAD

**Issue by wlandau-lilly** [wlandau-lilly/drake#47]
Base 64 encoding uses both uppercase and lowercase letters, and I did not realize until now that this behavior could cause collisions in case-insensitive file systems like Windows. The risk is low, but it is enough to switch from `base64url::base64_urlencode()` to `base64url::base32_encode()` to compute the names of timestamp files. This only affects how Makefiles are written and executed, and a new Makefile is generated on every call to `drake::make(..., parallelism = "Makefile")`, so this fix will not affect back compatibility. Existing projects will be unharmed.

**Comment by wlandau-lilly** [comment]
Base 32 encoding produces long names, and the length is variable. This may be a problem because Windows file paths cannot be longer than 260 characters. Let's use hashing instead of base 32 encoding. Since a new `Makefile` is generated on every call to `make(..., parallelism = "Makefile")`, the change will be totally back-compatible. And as far as I can tell, all the letters in the hashes are lower case, so case-insensitivity on Windows  will not be a problem.

## Changes planned

- Use hashing in [safe_encode()](https://github.com/wlandau-lilly/drake/blob/master/R/timestamp.R#L11).
```r
safe_encode <- function(x){
  digest(x, algo = hash_algorithm)
}
```

- Remove `base64url` [from the Imports: list in the DESCRIPTION](https://github.com/wlandau-lilly/drake/blob/master/DESCRIPTION#L30).
- Remove `base64url` [from the roxygen2 comments in package.R](https://github.com/wlandau-lilly/drake/blob/master/R/package.R#L30).
- Regenerate the documentation with `devtools::document()`
- To be safe, test `Makefile` parallelism by running [multiple.R](https://github.com/wlandau-lilly/drake/blob/master/tests/extra/multiple.R) on Linux and Windows. (It should only take one coding session to fix all the [easy issues](https://github.com/wlandau-lilly/drake/labels/easy), so we can probably wait until then to do the time-consuming tests.)

**Comment by wlandau-lilly** [comment]
This is actually turning out to be difficult to implement because I am also working on #53 simultaneously in the `hash` branch. The main tricky part is exposing the hash algorithms as arguments to `make()` while respecting back compatibility with old caches.

**Comment by wlandau-lilly** [comment]
Fixed by 50072106d6c7bea924bd67f21bbe0b0e44b25b2e.
